File: ALifeCommon/src/main/java/com/cristiano/alife/viewer/IViewLife.java
package com.cristiano.alife.viewer;

import java.awt.Color;

import com.cristiano.alife.world.IOrganismo;

public interface IViewLife {

	void drawRect(Color cor, int x, int y, int graphWidth, int graphWidth2);

	void showDetails(IOrganismo o);

	void repaint();

	void checkTick();

	void transpRect(int i, int j, int p, int p2);

	void line(int xp, int yp, int i, int j);

	int selectedOrgId();

	void drawCircle(Color red, int x, int y, int i, int j);

	void drawLine(Color cor, int x1, int y1, int x2, int y2);

	void setSelected(IOrganismo o);

}

File: netAvida/netAvida.Backend/interfaces/IViewLife.cs
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.interfaces
{
    public interface IViewLife
    {
        void DrawRect(int value, int x, int y, int graphWidth, int graphWidth2);

        void ShowDetails(IOrganismo o);

        void Update();

        void CheckTick();

        void TranspRect(int i, int j, int p, int p2);

        int SelectedOrgId();

        void DrawCircle(int value, int x, int y, int i, int j);

        void DrawLine(int value, int x1, int y1, int x2, int y2);
        void SetMundo(IWorld world);
        void SetSelected(IOrganismo o);

        
    }
}
